03-rnaseq/differential-expression_rnaseq_01.Rmd

```diff
@@ -0,0 +1,417 @@
+TODO: Add a note/link to section in documentation on why we use `rlog` vs `vst` and `normTransform`.
+
+```{r}
+# Normalize the data in the `DESeqDataSet` object using the `rlog()` function from the `DESEq2` R package
+ddset_norm <- rlog(ddset)
+```
+
+## Run differential expression analysis
+
+We'll use the wrapper function `DESeq()` to do our differential expression analysis.
+
+```{r}
+deseq_object <- DESeq(ddset)
+```
+
+Let's take a look at the results table.
+
+```{r}
+deseq_results <- results(deseq_object)
+deseq_results
+```
+
+Sort and filter the results. 
+
+```{r}
+# this is of class DESeqResults -- we want a data frame
+deseq_df <- deseq_results %>%
+  # make into data.frame
+  as.data.frame() %>%
+  # the gene names are rownames -- let's make this it's own column for easy
+  # display
+  tibble::rownames_to_column("Gene") %>%
+  dplyr::mutate(threshold = padj < 0.05) %>%
+  # let's sort by statistic -- the highest values should be what is up in the
+  # ASX mutated samples
+  dplyr::arrange(dplyr::desc(stat))
+```
+
+Let's print out what the top results are. 
+
+```{r}
+head(deseq_df)
+```
+
+### Check results by plotting one gene
+
+To double check what a differentially expressed gene looks like, we can plot one with `DESeq2::plotCounts()` function. 
+
+```{r}
+plotCounts(ddset, gene = "ENSG00000196074", intgroup = "asxl_mutation_status")
+```
+The `mutation` group samples have higher expression of this gene than the control group, which helps assure us that the results are showing us what we are looking for. 
+
+## Save results to TSV 
+
+Write the results table to file.
+
+```{r}
+readr::write_tsv(
+  deseq_df,
+  file.path(
+    results_dir,
+    "SRP078441_differential_expression_results.tsv" # Replace with a relevant output file name
+  )
+)
+```
+
+## Create a volcano plot
+
+```{r}
+volcano_plot <- ggplot(deseq_df, aes(x = log2FoldChange, y = -log10(padj), color = threshold)) +
+  geom_point() +
+  theme_classic() +
+  xlab("log2 fold change") +
+  ylab("-log10 adjusted p-value") +
+  ggtitle("Volcano plot for mutation status") +
+  scale_color_manual(values = c("lightgray", "navy")) +
+  theme(legend.position = "none")
+
+# Print out plot here
+volcano_plot
+```
+
+Save the plot to a PNG. 
+
+```{r}
+ggsave(
+  plot = volcano_plot,
+  file.path(plots_dir, "SRP078441_volcano_plot.png")
+) # Replace with a plot name relevant to your data
+```
+
+Heatmaps are also a pretty common way to show differential expression results. 
+You can take your results from this example and make a heatmap following our heatmap module: TODO: link here. 
+
+# Further learning resources about this analysis
+
+- [`DESeq2` vignette](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html)
+- [DESeq2 paper](https://doi.org/10.1186/s13059-014-0550-8) [@Love]
+- [StatQuest Video: DESeq2, part 1, Library Normalization](https://www.youtube.com/watch?v=UFB993xufUU)
+
+# Print session info:
```

Comment:
I think some of these have not been updated to reflected the new template yet (apologies if that has already been discussed as part of this) review

https://github.com/AlexsLemonade/refinebio-examples/blob/a6ee8c92038972d990d3009f870a18e90ab3819d/template/template_example.Rmd#L217

Reply:
Yes, I need to take another pass through the template parts since pieces have been updated since when this PR was last reviewed.